ddtrace/tracer/span.go

```diff
@@ -169,6 +170,37 @@ func (s *span) setSamplingPriority(priority int, sampler samplernames.SamplerNam
 	s.setSamplingPriorityLocked(priority, sampler, rate)
 }
 
+// setUser sets the span user ID tag as well as some optional user monitoring tags depending on the configuration.
+// The function assumes that the span it is called on is the trace's root span.
+func (s *span) setUser(id string, cfg UserMonitoringConfig) {
+	tags := map[string]string{}
+	keys := []string{ext.UserEmail, ext.UserName, ext.UserScope, ext.UserRole, ext.UserSessionID}
+	vals := []string{cfg.email, cfg.name, cfg.scope, cfg.role, cfg.sessionID}
+	// Build optional tags map before locking
+	for i, v := range vals {
+		if v != "" {
+			tags[keys[i]] = v
+		}
+	}
+	s.Lock()
+	defer s.Unlock()
+	if cfg.propagateID {
+		// Delete usr.id from the tags since _dd.p.usr.id takes precedence
+		delete(s.Meta, ext.UserID)
+		id = base64.StdEncoding.EncodeToString([]byte(id))
```

Comment:
I don't think it's a good practice to override the function's parameter. Maybe you can chance to
```suggestion
		idenc := base64.StdEncoding.EncodeToString([]byte(id))
```
As in "ID encoded"